gearcmd/worker.go

```diff
@@ -302,7 +302,14 @@ func stopProcess(p *os.Process, gracePeriod time.Duration) error {
 		}
 		time.Sleep(gracePeriod)
 	}
-	if err := syscall.Kill(p.Pid, syscall.SIGKILL); err != nil {
+	// kill entire group of process spawned by our cmd.Process
+	pgid, err := syscall.Getpgid(p.Pid)
+	if err != nil {
+		return fmt.Errorf("unable to get pgid, error: %s", err)
```

Comment:
I think we should not fail here, why not log and then still kill the base process?

This is how I had started to do it:
```diff
diff --git a/gearcmd/worker.go b/gearcmd/worker.go
index eaccec5..e48c57c 100644
--- a/gearcmd/worker.go
+++ b/gearcmd/worker.go
@@ -302,7 +302,14 @@ func stopProcess(p *os.Process, gracePeriod time.Duration) error {
                }
                time.Sleep(gracePeriod)
        }
-       if err := syscall.Kill(p.Pid, syscall.SIGKILL); err != nil {
+
+       targetID := p.Pid
+       pgid, err := syscall.Getpgid(p.Pid)
+       if err == nil {
+               targetID = -pgid
+       }
+
+       if err := syscall.Kill(targetID, syscall.SIGKILL); err != nil {
                // The graceful shutdown may have completed, so we don't error in that case.
                if gracePeriod == 0 || !strings.Contains(err.Error(), "no such process") {
                        return fmt.Errorf("unable to send SIGKILL, error: %s", err)
```

Reply:
👍